apps/main/src/user/server/user.ts

```diff
@@ -34,35 +34,43 @@ export const getUserContext = async (): Promise<UserContextValue> => {
   return { authUser, user };
 };
 
-export const verifyUserContext = async () => {
+export const verifyUser = async (
+  options: {
+    requireEmailVerified?: boolean;
+    requireProfile?: boolean;
+  } = {},
+) => {
+  const { requireEmailVerified = true, requireProfile = true } = options;
+
   const authUser = await getAuthUser();
+
   if (!authUser) {
     redirect('/login');
   }
-  const user = await getUser();
-  if (!user) {
-    redirect('/new-user');
+
+  if (requireEmailVerified && !authUser.emailVerified) {
+    redirect('/verify-email');
   }
-  return { uid: authUser.uid, authUser, user };
-};
 
-export const validUserContext = async () => {
-  // A user may either be authorized and have a user profile; or they must be
-  // unauthorized.
-  const authUser = await getAuthUser();
   const user = await getUser();
-  if (authUser && user) {
-    return { authUser, user };
-  } else if (authUser && !user) {
+
+  if (requireProfile && !user) {
     redirect('/new-user');
-  } else if (!authUser && user) {
-    // This can't ever really happen, cause getUser calls getAuthUser.
-    throw Error('Unexpected user state!');
-  } else {
-    return { authUser, user };
   }
+
+  return { authUser, user };
 };
 
+/**
+ * @deprecated Use `verifyUser` instead.
```

Comment:
this is creating techincal debt. please restore these functions, internally passing in the correct values if you need to.